report/controls.go

```diff
@@ -84,25 +83,20 @@ type WireNodeControls struct {
 	Controls  StringSet `json:"controls,omitempty"`
 }
 
-// MarshalJSON implements json.Marshaller
-func (nc NodeControls) MarshalJSON() ([]byte, error) {
-	buf := bytes.Buffer{}
-	err := json.NewEncoder(&buf).Encode(WireNodeControls{
+func (nc *NodeControls) CodecEncodeSelf(encoder *codec.Encoder) {
+	encoder.Encode(WireNodeControls{
 		Timestamp: renderTime(nc.Timestamp),
 		Controls:  nc.Controls,
 	})
-	return buf.Bytes(), err
 }
 
-// UnmarshalJSON implements json.Unmarshaler
-func (nc *NodeControls) UnmarshalJSON(input []byte) error {
+func (nc *NodeControls) CodecDecodeSelf(decoder *codec.Decoder) {
```

Comment:
We shouldn't use `json.Decode()` anymore. Tests-aside, I thought I had removed all references of `encoding/json`. Where did you see that?